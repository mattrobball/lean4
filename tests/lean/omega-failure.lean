/-!
Testing omega's failure message
-/

example : 0 < 0 := by omega

example (x : Nat) : x < 0 := by omega

example (x : Nat) (y : Int) : x < y := by omega

example (x y : Int) : 5 < x ∧ x < 10 → y > 0 := by omega

-- this used to fail with y = x, but then omega got better, so now there are unrelated x and y
-- to make omega fail
theorem sizeOf_snd_lt_sizeOf_list {α : Type u} {β : Type v} [SizeOf α] [SizeOf β]
  {x y : α × β} {xs : List (α × β)} :
  y ∈ xs → sizeOf x.snd < 1 + sizeOf xs
:= by
  intro h
  have := List.sizeOf_lt_of_mem h
  have : sizeOf x = 1 + sizeOf x.1 + sizeOf x.2 := rfl
  omega


example (reallyreallyreallyreally longlonglonglong namenamename : Nat) :
  reallyreallyreallyreally < longlonglonglong + namenamename := by omega


def a := 1

example (b c d e f g h i j k l m n o p : Nat) :
  b + c + d + e + f + g + h + i + j + k + l + m + n + o + p < 100 := by omega
