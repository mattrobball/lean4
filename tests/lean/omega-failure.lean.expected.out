omega-failure.lean:5:22-5:27: error: omega could not prove the goal:
it is false
omega-failure.lean:7:32-7:37: error: omega could not prove the goal:
a possible counterexample may satisfy the constraints
  a ≥ 0
where
 a := ↑x
omega-failure.lean:9:42-9:47: error: omega could not prove the goal:
a possible counterexample may satisfy the constraints
  a - b ≥ 0
  a ≥ 0
where
 a := ↑x
 b := y
omega-failure.lean:11:51-11:56: error: omega could not prove the goal:
a possible counterexample may satisfy the constraints
  b ≤ 0
  6 ≤ a ≤ 9
where
 a := x
 b := y
omega-failure.lean:22:2-22:7: error: omega could not prove the goal:
a possible counterexample may satisfy the constraints
  d ≥ 0
  a - b ≥ 1
  c ≥ 0
  a - d ≤ -1
  b ≥ 0
  a ≥ 0
  c + d ≥ -1
where
 a := ↑(sizeOf xs)
 b := ↑(sizeOf y)
 c := ↑(sizeOf x.fst)
 d := ↑(sizeOf x.snd)
omega-failure.lean:26:67-26:72: error: omega could not prove the goal:
a possible counterexample may satisfy the constraints
  c ≥ 0
  a - b - c ≥ 0
  b ≥ 0
  a ≥ 0
where
 a := ↑reallyreallyreallyreally
 b := ↑longlonglonglong
 c := ↑namenamename
omega-failure.lean:32:72-32:77: error: omega could not prove the goal:
a possible counterexample may satisfy the constraints
  x ≥ 0
  a_1 ≥ 0
  v ≥ 0
  c_1 ≥ 0
  b_1 ≥ 0
  e_1 ≥ 0
  z ≥ 0
  q ≥ 0
  s ≥ 0
  d_1 ≥ 0
  u ≥ 0
  t ≥ 0
  w ≥ 0
  q + r + s + t + u + v + w + x + y + z + a_1 + b_1 + c_1 + d_1 + e_1 ≥ 100
  r ≥ 0
  y ≥ 0
where
 q := ↑b
 r := ↑c
 s := ↑d
 t := ↑e
 u := ↑f
 v := ↑g
 w := ↑h
 x := ↑i
 y := ↑j
 z := ↑k
 a_1 := ↑l
 b_1 := ↑m
 c_1 := ↑n
 d_1 := ↑o
 e_1 := ↑p
